diff --git a/widget/headless/HeadlessWidget.h b/widget/headless/HeadlessWidget.h
--- a/widget/headless/HeadlessWidget.h
+++ b/widget/headless/HeadlessWidget.h
@@ -16,6 +16,10 @@
 #  define MOZ_HEADLESS_SCROLL_MULTIPLIER 3
 #  define MOZ_HEADLESS_SCROLL_DELTA_MODE \
     mozilla::dom::WheelEvent_Binding::DOM_DELTA_LINE
+#elif defined(MOZ_WIDGET_MAKOS)
+#  define MOZ_HEADLESS_SCROLL_MULTIPLIER 3
+#  define MOZ_HEADLESS_SCROLL_DELTA_MODE \
+    mozilla::dom::WheelEvent_Binding::DOM_DELTA_LINE
 #elif defined(XP_WIN)
 #  define MOZ_HEADLESS_SCROLL_MULTIPLIER \
     .025  // default scroll lines (3) / WHEEL_DELTA (120)
