diff --git a/src/unix.rs b/src/unix.rs
--- a/src/unix.rs
+++ b/src/unix.rs
@@ -87,10 +87,11 @@ extern "C" {
     #[cfg_attr(
         any(
             target_os = "linux",
+            target_os = "makos",
             target_os = "hurd",
             target_os = "redox",
             target_os = "dragonfly"
         ),
         link_name = "__errno_location"
     )]
     #[cfg_attr(target_os = "aix", link_name = "_Errno")]
