diff --git a/config.sub b/config.sub
--- a/config.sub
+++ b/config.sub
@@ -1743,7 +1743,7 @@ case $os in
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]* \
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
-	     | os9* | macos* | osx* | ios* | tvos* | watchos* \
+	     | os9* | macos* | makos* | osx* | ios* | tvos* | watchos* \
 	     | mpw* | magic* | mmixware* | mon960* | lnews* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
 	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -332,6 +332,9 @@ if test -z "$MACHDEP"
 	*-*-darwin*)
 		ac_sys_system=Darwin
 		;;
+	*-*-makos*)
+		ac_sys_system=MakOS
+		;;
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
@@ -803,6 +806,9 @@ if test "$cross_compiling" = yes; then
 		*)
 			_host_ident=$host_cpu
 		esac
 		;;
+	*-*-makos*)
+		_host_ident=$host_cpu
+		;;
 	*-*-vxworks*)
 		_host_ident=$host_cpu
 		;;
diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -4120,6 +4120,9 @@ then :
 	*-*-darwin*)
 		ac_sys_system=Darwin
 		;;
+	*-*-makos*)
+		ac_sys_system=MakOS
+		;;
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
@@ -4624,6 +4627,9 @@ then :
 		*)
 			_host_ident=$host_cpu
 		esac
 		;;
+	*-*-makos*)
+		_host_ident=$host_cpu
+		;;
 	*-*-vxworks*)
 		_host_ident=$host_cpu
 		;;
