diff --git a/src/internal/makos_syscall.c b/src/internal/makos_syscall.c
--- a/src/internal/makos_syscall.c
+++ b/src/internal/makos_syscall.c
@@ -19,5 +19,6 @@ enum linux_aarch64_syscall {
 	L_read = 63,
 	L_write = 64,
+	L_set_tid_address = 96,
 	L_exit = 93,
 	L_exit_group = 94,
 	L_clock_gettime = 113,
@@ -28,5 +29,6 @@ enum linux_aarch64_syscall {
 	L_setpgid = 154,
 	L_getpgid = 155,
+	L_gettid = 178,
 	L_socket = 198,
 	L_connect = 203,
 	L_sendto = 206,
@@ -66,4 +68,6 @@ enum makos_syscall {
 	M_tcsetpgrp = 74,
 	M_brk = 76,
 	M_rename = 77,
+	M_set_tid_address = 78,
+	M_gettid = 79,
 };
@@ -282,5 +286,9 @@ hidden long __makos_syscall_dispatch(long n, long a, long b, long c,
 		result = makos_raw(M_brk, a, 0, 0, 0, 0, 0);
 		return result == -1 ? 0 : result;
+	case L_set_tid_address:
+		return makos_raw(M_set_tid_address, a, 0, 0, 0, 0, 0);
+	case L_gettid:
+		return makos_raw(M_gettid, 0, 0, 0, 0, 0, 0);
 	case L_exit:
 	case L_exit_group:
 		makos_raw(M_exit, a, 0, 0, 0, 0, 0);
