diff --git a/src/internal/makos_syscall.c b/src/internal/makos_syscall.c
--- a/src/internal/makos_syscall.c
+++ b/src/internal/makos_syscall.c
@@ -17,6 +17,7 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#include <sys/prctl.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <termios.h>
@@ -65,6 +66,7 @@ enum linux_aarch64_syscall {
 	L_rt_sigprocmask = 135,
 	L_setpgid = 154,
 	L_getpgid = 155,
+	L_prctl = 167,
 	L_getpid = 172,
 	L_getppid = 173,
 	L_getuid = 174,
@@ -159,6 +161,7 @@ enum makos_syscall {
 	M_pselect = 109,
 	M_execve = 112,
 	M_madvise = 114,
+	M_thread_set_name = 115,
 };
 
 struct makos_metadata {
@@ -810,6 +813,11 @@ hidden long __makos_syscall_dispatch(long n, long a, long b, long c,
 		return makos_bool_result(makos_raw(M_mprotect, a, b, c, 0, 0, 0), EINVAL);
 	case L_madvise:
 		return makos_raw(M_madvise, a, b, c, 0, 0, 0);
+	case L_prctl:
+		if (a != PR_SET_NAME || !b) return -EINVAL;
+		result = strnlen((const char *)b, 16);
+		if (result == 16) return -ERANGE;
+		return makos_raw(M_thread_set_name, b, result, 0, 0, 0, 0);
 	case L_brk:
 		result = makos_raw(M_brk, a, 0, 0, 0, 0, 0);
 		return result == -1 ? 0 : result;
